tests/product_tests/foreman/foreman_tests.py

```diff
@@ -0,0 +1,100 @@
+
+
+class ForemanBasicTest(StageOneReportTest):
+    """Ensure that a basic execution runs as expected with all TFM related
+    plugins. For the Red Hat tests, it assumes Foreman has been deployed on a
+    Katello system. On Debian systems, these tests are skipped.
+
+    :avocado: tags=foreman
+    """
+
+    sos_cmd = '-v'
+
+    def test_tfm_plugins_ran(self):
+        self.assertPluginIncluded([
+            'apache',
+            'foreman',
+            'postgresql',
+            'puppet',
+            'ruby'
+        ])
+
+    @redhat_only
+    def test_candlepin_plugin_ran(self):
+        self.assertPluginIncluded('candlepin')
+
+    def test_foreman_keys_skipped(self):
+        self.assertFileGlobNotInArchive("/etc/foreman*/*key.pem")
+
+    def test_foreman_database_sizes_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_db_tables_sizes')
+
+    def test_foreman_installer_dirs_collected(self):
+        self.assertFileGlobInArchive("/etc/foreman-installer/*")
+        self.assertFileGlobInArchive("/var/log/foreman-installer/*")
+
+    def test_foreman_production_log_collected(self):
+        self.assertFileCollected('/var/log/foreman/production.log')
+
+    def test_foreman_database_dumps_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_settings_table')
+
+    def test_foreman_csv_collected(self):
+        self.assertFileCollected('sos_commands/foreman/foreman_tasks_tasks')
+
+    def test_proxyfeatures_not_collected(self):
+        self.assertFileGlobNotInArchive("*/smart_proxies_features/*")
+
+    def test_foreman_config_postproc_worked(self):
+        self.assertFileNotHasContent('/etc/foreman/database.yml', FOREMAN_DB_PASSWORD)
+
+    def test_foreman_password_postproc_worked(self):
+        for passwd in FOREMAN_PASSWORDS:
+            self.assertFileNotHasContent('/var/log/foreman-installer/foreman.log', passwd)
+
+    @redhat_only
+    def test_candlepin_table_sizes_collected(self):
+        self.assertFileCollected('sos_commands/candlepin/candlepin_db_tables_sizes')
+
+    @redhat_only
+    def test_katello_password_postproc_worked(self):
+        for _check in ['/var/log/foreman-installer/katello.log', '/etc/foreman-installer/scenarios.d/katello-answers.yaml']:
+            for passwd in FOREMAN_PASSWORDS:
+                self.assertFileNotHasContent(_check, passwd)
+
+    @redhat_only
+    def test_foreman_httpd_collected(self):
+        self.assertFileGlobInArchive("/var/log/httpd*/foreman-ssl_*_ssl*log*")
+
+    def test_ruby_gems_collected(self):
+        self.assertFileCollected('sos_commands/ruby/gem_list')
+
+
+class ForemanWithOptionsTest(StageOneReportTest):
+    """Enable Foreman/Katello/Candlepin specific options and ensure they are
+    working
+
+    :avocado: tags=foreman
+    """
+
+    sos_cmd = '-v -k foreman.proxyfeatures=on'
+
+    def test_proxyfeatures_collected(self):
+        self.assertFileGlobInArchive("*/smart_proxies_features/*")
```

Comment:
Looking at the code, I'd expect this to fail on non-Katello installs:
https://github.com/sosreport/sos/blob/61ff5ce165e654a02fe80b9de5ec8e49ed808ec9/sos/report/plugins/foreman.py#L256-L257

Those paths do not exist in non-katello. Technically you need to check it in the database but in most installations you can find `ssl_certificate`, `ssl_ca_file` and `ssl_priv_key` in `/etc/foreman/settings.yaml`.

If this test passes, then perhaps it should also verify that it at least got data for one proxy.

Reply:
That's my fault, that test actually needs to have the plugin dir in the glob check, but also yeah that needs to be tagged `redhat_only`

Reply:
If you follow my suggestion, then you use the configured certificates and it would also work on Debian. Reading something that's configurable from the config files feels more correct to me.

Reply:
Oh, I follow now.

For the moment I'll leave that since @pmoravec is planning on splitting out proxy stuff into a separate plugin. We'll update that check there and then can extend the relevant test(s) back out to debian.

Reply:
That's fair, it's really a bugfix separate from CI testing.